### PR TITLE
Shorten the conversation tests for speed + fixing position overflows

### PR DESCRIPTION
**Pull request by Rocketknight1**
Some of the conversation tests were still overflowing the maximum position embeddings for very short models like `BlenderBotSmall`. I reduced the limits a lot, which also speeds up those tests! cc @ydshieh

### CODE REVIEW

**Comment by Rocketknight1** [comment]
@ydshieh the test never failed before because the `ConversationalPipeline` class wasn't working very well, and didn't actually generate any text in these tests!

**Comment by ydshieh** [comment]
Thanks. Then let's just skip it for this model. You can search `def is_pipeline_test_to_skip` to see how to skip pipeline tests.

**Comment by Rocketknight1** [comment]
@ydshieh are you sure? I think the test makes more sense with very short outputs anyway - we don't need 20 tokens to confirm the pipeline works!

**Comment by ydshieh** [comment]
That's true, but making it this way also hide the fact of some models fail this test. Having skip make it explicit. We can definitely use `2` instead of `20` if we do have a gain of running time (not that usually the model used for pipeline testing is very small, and not really obvious to say we will have a gain).

**Comment by Rocketknight1** [comment]
BlenderBotSmall can pass the test! The problem only arises because the test `BlenderBotSmallConfig` has `max_position_embeddings=20`. Maybe I keep the test at a higher number, but increase the `BlenderBotSmallConfig` in the test to have more position embeddings, so it passes more easily?

**Comment by ArthurZucker** [comment]
Hey! Let's just skip it for now it's blocking a few prs!

**Comment by ArthurZucker** [comment]
I'll merge #27013 so you have time to work on the fix

**Comment by Rocketknight1** [comment]
cc @ydshieh @ArthurZucker this should be ready to go now! I've skipped the test for `BlenderBotSmall` as @ydshieh suggested because the problem is in the `hf-internal-testing` checkpoint, and I don't think the model has much usage anymore. The model actually works fine - the test checkpoint just has a very low value for `max_position_embeddings`.

I also increased `max_new_tokens` back to `5` in the general test and removed the workarounds that @ArthurZucker added. I don't think it needs to go higher - it just makes the test slower, and doesn't tell us anything new!

**Comment by Rocketknight1** [comment]
Quick ping again @ydshieh @ArthurZucker! This PR should be ready to go

**Comment by ydshieh** [comment]
Damm, I made review but forgot to commit the comments. Sorry

File: tests/models/blenderbot/test_modeling_blenderbot.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def __init__(
         hidden_act="gelu",
         hidden_dropout_prob=0.1,
         attention_probs_dropout_prob=0.1,
-        max_position_embeddings=20,
+        max_position_embeddings=50,
```
[the remark this answers is not in the document]
Comment (ydshieh) [reply tests/models/blenderbot/test_modeling_blenderbot.py:88]: (the above is just a comment, not a request to change)

**Review by ydshieh — APPROVED** [review]

File: tests/models/blenderbot_small/test_modeling_blenderbot_small.py [hunk 2]
```diff
@@ -284,6 +281,10 @@ def test_generate_fp16(self):
         model.generate(input_ids, attention_mask=attention_mask)
         model.generate(num_beams=4, do_sample=True, early_stopping=False, num_return_sequences=3)
 
+    @unittest.skip("Tiny random model has too few position embeddings for this.")
+    def test_pipeline_conversational(self):
+        pass
```
[the remark this answers is not in the document]
Comment (Rocketknight1) [reply tests/models/blenderbot_small/test_modeling_blenderbot_small.py:286]: Will do!

**Comment by Rocketknight1** [comment]
@ydshieh Changes made as you suggested - I used `is_pipeline_test_to_skip`. I left the updated sequence lengths alone because even though they don't fix our `tiny-random` models, they do fix some other tests, especially for `BlenderBot`.

File: tests/models/blenderbot_small/test_modeling_blenderbot_small.py [hunk 3]
```diff
@@ -242,10 +242,7 @@ class BlenderbotSmallModelTest(ModelTesterMixin, GenerationTesterMixin, Pipeline
     def is_pipeline_test_to_skip(
         self, pipeline_test_casse_name, config_class, model_architecture, tokenizer_name, processor_name
     ):
-        if pipeline_test_casse_name == "TextGenerationPipelineTests":
-            return True
-        # TODO @Rocketnight1 to fix
-        if pipeline_test_casse_name == "ConversationalPipelineTests":
+        if pipeline_test_casse_name in ("TextGenerationPipelineTests", "ConversationalPipelineTests"):
             return True
         return False
```
Comment (amyeroberts) [inline tests/models/blenderbot_small/test_modeling_blenderbot_small.py:247]:
```suggestion
        return pipeline_test_casse_name in ("TextGenerationPipelineTests", "ConversationalPipelineTests")
```

File: tests/models/blenderbot_small/test_modeling_flax_blenderbot_small.py [hunk 4]
```diff
@@ -320,6 +320,13 @@ class FlaxBlenderbotSmallModelTest(FlaxModelTesterMixin, unittest.TestCase, Flax
     )
     all_generative_model_classes = (FlaxBlenderbotSmallForConditionalGeneration,) if is_flax_available() else ()
 
+    def is_pipeline_test_to_skip(
+        self, pipeline_test_casse_name, config_class, model_architecture, tokenizer_name, processor_name
+    ):
+        if pipeline_test_casse_name in ("TextGenerationPipelineTests", "ConversationalPipelineTests"):
+            return True
+        return False
```
Comment (amyeroberts) [inline tests/models/blenderbot_small/test_modeling_flax_blenderbot_small.py:328]:
```suggestion
        return pipeline_test_casse_name in ("TextGenerationPipelineTests", "ConversationalPipelineTests")
```

File: tests/models/blenderbot_small/test_modeling_tf_blenderbot_small.py [hunk 5]
```diff
@@ -198,6 +198,13 @@ class TFBlenderbotSmallModelTest(TFModelTesterMixin, PipelineTesterMixin, unitte
     test_pruning = False
     test_onnx = False
 
+    def is_pipeline_test_to_skip(
+        self, pipeline_test_casse_name, config_class, model_architecture, tokenizer_name, processor_name
+    ):
+        if pipeline_test_casse_name in ("TextGenerationPipelineTests", "ConversationalPipelineTests"):
+            return True
+        return False
```
Comment (amyeroberts) [inline tests/models/blenderbot_small/test_modeling_tf_blenderbot_small.py:206]:
```suggestion
        return pipeline_test_casse_name in ("TextGenerationPipelineTests", "ConversationalPipelineTests")
```

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!

File: tests/models/blenderbot_small/test_modeling_blenderbot_small.py (hunk 3 above)
Comment (ydshieh) [reply tests/models/blenderbot_small/test_modeling_blenderbot_small.py:247]: @amyeroberts That was putting that way for 2 (minor) reasons:

  - it's likely the condition could get more branches: if further failing tests happens or new pipeline test classes are added
  - make it clearly that, by default, it should return `False`: so no test should be skipped

But good for me if you prefer to apply the suggested changes

Comment (amyeroberts) [reply tests/models/blenderbot_small/test_modeling_blenderbot_small.py:247]: I don't feel very strongly - they were just suggestions. Happy to leave as-is :)

Comment (Rocketknight1) [reply tests/models/blenderbot_small/test_modeling_blenderbot_small.py:247]: I like the suggestion, so I'll take it! We can always edit later if we need to change the condition.

**Pull request merged**